### ISSUE TITLE
Digital Ocean and ShadowSocks

### ISSUE THREAD

**Issue by johnbeynon** [jlund/streisand#558]
I've just built my first Streisand Server using Digital Ocean. I've confirmed that L2TP and Cisco AnyConnect work but ShadowSocks (Android) does not work, it connects and I  see bytes sent but I get nothing returned. Is this related to UDP 80/443 on DO perhaps?

**Comment by swinchen** [comment]
Same issue here.  I have verified that WireGuard and OpenVPN work, but Shadowsocks (Android) does not work.

**Comment by denoza** [comment]
I have a guess, have you checked your shadowsocks android client's version? Perhaps the new versions don't support OTA which is deprecated in the newer versions of shadowsocks. Since, streisand still uses deprecated OTA therefore client's packets might be rejected by the server. I am not sure about it, I have already shifted to Shadowsocks R for my android traffic.

**Comment by KickCellarDoor** [comment]
Same issue. But I was using the default shadowsocksX-ng client for Mac.

**Comment by mveplus** [comment]
Hello guys, 
after a few days redeploying a few times on DO, I've looked here and found I'm not alone ;) 
I've tried it on iOS, Android, Windows, OS X - shadowsocks does not work with or without OTA. 
I've looked the version deployed by ansible and it was not latest one - shadowsocks-libev 2.5.6 with OpenSSL 1.0.2g  1 Mar 2016. Ansible compile it by source during the installation so it's strange why it deploys older shadowsocks-libev when a current one is v.3.0.5? 
I've done the update and a removed "auth":"true" in config and I had to disable auth in the client as well, so now it does connect but does have some problems with routeing the packages, I'll investigate that further. 

Greetings, 

P.S. I've up-to-date Streisand from GitHub.

**Comment by ortonomy** [comment]
By the way, Shadowsocks-NG 1.5 for mac has completely deprecated OTA. If you look at the logs ``ss-local`` it says 'ERROR, OTA has been deprecated``

This is in ``v1.5`` only.

**Comment by ortonomy** [comment]
@johnbeynon - two possible fixes

1. The selected encryption method in Shadowssocks-NG when you import from QR code is sometimes incorrect. Streisand selects ``aes-256-cfb`` as the cipher, but SS-NG imports it as a captilized version with ``-auth`` appended. I changed this back to the correct string.
2. The local listen port for the SOCKS5 proxy should also be set to ``1080``. 

It was these two settings that fixed my issues.

**Comment by johnbeynon** [comment]
I'm using ShadowSocks on Android to connect, not Shadowssocks-NG.

**Comment by tiliarou** [comment]
Shadowsock (not NG) since 4.0 has deprecated OTA.

**Comment by cpu** [comment]
I recommend folks follow https://github.com/jlund/streisand/issues/586 for the upgrade. Once https://github.com/jlund/streisand/pull/611 is merged I will likely close this issue.

**Comment by cpu** [comment]
#611 is merged - I'm going to close this issue. If you provision a new Streisand server using the latest code from master with the updated shadowsocks-libev server & configuration and you still are having trouble please open a new issue and I will help you debug.

Thanks!
